### PR TITLE
feat: add cursor:pointer to table if on row.onClick() implementation exists

### PR DESCRIPTION
**Pull request by SimBig**

### CODE REVIEW

File: src/Table/index.stories.tsx [hunk 1]
```diff
@@ -41,22 +41,31 @@ const data: Array<Person> = [
...
+Default.args = {
+  dataSource: data,
+  columns: columns,
+};
+
+export const TableWithOnRow = Template.bind({});
+TableWithOnRow.args = {
+  ...Default.args,
+  onRow: () => ({
+      onClick: () => {}, // click row
```
Comment (spawnia) [inline src/Table/index.stories.tsx]:
```suggestion
      onClick: () => {},
```

File: src/Table/index.stories.tsx [hunk 2]
```diff
@@ -41,22 +41,31 @@ const data: Array<Person> = [
...
   {
     title: 'Age',
     dataIndex: 'age',
   },
 ];
 
-export const Default: Story<TableProps<Person>> = (args) => (
-  <Table dataSource={data} columns={columns} rowKey="id" {...args} />
-);
+const Template: Story<TableProps<any>> = (args) => <Table {...args} />;
```
Comment (spawnia) [inline src/Table/index.stories.tsx:55]:
```suggestion
const Template: Story<TableProps<Person>> = (args) => <Table {...args} />;
```

File: src/Table/index.stories.tsx [hunk 3]
```diff
@@ -41,22 +41,31 @@ const data: Array<Person> = [
...
 
-const FilterDropdown: ColumnProps<Person>['filterDropdown'] = () => (
-  <div>test</div>
-);
-
 const columns: Array<ColumnProps<Person>> = [
   {
     title: 'Name',
     dataIndex: 'name',
-    filterDropdown: FilterDropdown,
```
Comment (spawnia) [inline src/Table/index.stories.tsx:52]: Warum entfernt?

Comment (SimBig) [reply src/Table/index.stories.tsx:52]: The implementation was removed here. Thus, I do not see a benefit in keeping the visual representation in storybook.

File: src/Table/index.stories.tsx (hunk 2 above)
Comment (SimBig) [reply src/Table/index.stories.tsx:55]: Not sure how to solve this type issue then? Can you help me with this?
image

File: src/Table/index.stories.tsx (hunk 3 above)
Comment (spawnia) [reply src/Table/index.stories.tsx:52]: The workaround for an issue with antd was removed. The feature still exists, is useful and can be explored:

image

File: src/Table/index.stories.tsx (hunk 2 above)
Comment (spawnia) [reply src/Table/index.stories.tsx:55]: Please don't ignore issues or mask them by making the components usage less typesafe. We would have hit this problem once implementing the component in our code. Using proper types in tests validates the component types are well-defined.

Comment (SimBig) [reply src/Table/index.stories.tsx:55]: Ok, to me it looked like a storybook artefact due to the extraction of the template and not like a problem that would happen in the app code.. :-(

**Pull request merged**